### PR TITLE
add distribution gradle plugin

### PR DESCRIPTION
**Pull request by colesnodgrass**
## What
- fix gradle failing due to not being able to find `distTar`
   > Could not create task ':airbyte-cli:copyGeneratedTar'.
   >
   > Could not get unknown property 'distTar' for task ':airbyte-cli:copyGeneratedTar' of type org.gradle.api.tasks.Copy.

## How
- add the `distribution` plugin to the main `build.gradle` file

### CODE REVIEW

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

File: build.gradle [hunk 1]
```diff
@@ -29,6 +29,7 @@ plugins {
     id 'com.diffplug.spotless' version '6.7.1'
     id 'com.github.hierynomus.license' version '0.16.1'
     id 'com.github.spotbugs' version '5.0.6'
+    id 'distribution'
```
Comment (davinchia) [inline build.gradle:32]: can we add a comment specifying that this is for compatibility with https://github.com/airbytehq/airbyte/blob/master/build.gradle#L222 and can be removed when we refactor this?

**Comment by colesnodgrass** [comment]
@davinchia comment has been added

**Review by davinchia — APPROVED** [review]

**Comment by davinchia** [comment]
LGTM 🎸

**Pull request merged**